ui/src/components/workflow/WorkflowMetaDetails.js

```diff
@@ -74,4 +73,55 @@ class WorkflowMetaDetails extends Component {
     );
   }
 };
-export default connect(state => state.workflow)(WorkflowMetaDetails);
+
+class JSONTab extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingJSON: false,
+      wfs: this.props.wfs,
```

Comment:
Wfs will be null in this case. Passing initial state to a component as a prop is not the right way because the constructor is only called once before mounting the component, hence the component will only take the initial wfs (null in this case because its not yet loaded) and won't update afterwards. 
We should implement **componentWillReceiveProps()** method to handle this.